packages/visual-editor/src/hooks/useComponentProps.ts

```diff
@@ -357,3 +375,51 @@ const addExtraDropzonePadding = (padding: string) =>
 
 const isPercentValue = (value?: string | number) =>
   typeof value === 'string' && value.endsWith('%');
+
+const calculateDesignVariableValue = ({
+  variableName,
+  variableMapping,
+  node,
+  findNodeById,
+}: {
+  findNodeById: (nodeId?: string) => ExperienceTreeNode | null;
+  variableName: string;
+  variableMapping: DesignValue;
+  node: ExperienceTreeNode;
+}) => {
+  if (node.type === ASSEMBLY_BLOCK_NODE_TYPE) {
+    const patternId = node.data.pattern?.id;
+
+    const exposedProperyName = node['exposedPropertyNameToKeyMap'][variableName];
+    if (!exposedProperyName || !patternId) {
+      return variableMapping.valuesByBreakpoint;
+    }
+
+    const exposedVariableDefinition =
+      componentRegistry.get(patternId)?.definition.variables[exposedProperyName];
+
+    let exposedDefaultValue = exposedVariableDefinition?.defaultValue;
+    let parentPatternNode = findNodeById(node.data.pattern?.nodeId);
+
+    while (parentPatternNode) {
+      const parentPatternId = parentPatternNode.data.pattern?.id;
+      const nextKey = parentPatternNode['exposedPropertyNameToKeyMap'][exposedProperyName];
+
+      if (!parentPatternId || !nextKey) {
+        break;
+      }
+      const parentPatternVariableDefinition =
+        componentRegistry.get(parentPatternId)?.definition.variables[nextKey];
+
+      exposedDefaultValue = parentPatternVariableDefinition?.defaultValue;
```

Comment:
Might be worth noting down (in a comment) that with this logic, we're not merging defaults from different nesting layers for cases like "upper pattern specifies mobile and nested pattern specifies tablet".

Reply:
Should we handle the cases differently if there was no `defaultValue` for the property or are we populating the defaultValue on every level anyway in the web app?
If we didn't and the upper pattern had no default, we would not fall back to the nested definition.